Cabal/Distribution/Types/InstalledPackageInfo/FieldGrammar.hs

```diff
@@ -253,14 +265,17 @@ basicPkgName f b = (\x -> b { _basicPkgName = x }) <$> f (_basicPkgName b)
 {-# INLINE basicPkgName #-}
 
 basicLibName :: Lens' Basic (Maybe UnqualComponentName)
-basicLibName f b = (\x -> b { _basicLibName = x }) <$> f (_basicLibName b)
+basicLibName f b = (\x -> b { _basicLibName = maybeToLibraryName x }) <$>
+    f (libraryNameString (_basicLibName b))
 {-# INLINE basicLibName #-}
 
 basicFieldGrammar
     :: (FieldGrammar g, Applicative (g Basic))
     => g Basic Basic
-basicFieldGrammar = Basic
+basicFieldGrammar = mkBasic
     <$> optionalFieldDefAla "name"          MQuoted  basicName (mungedPackageName emptyInstalledPackageInfo)
     <*> optionalFieldDefAla "version"       MQuoted  basicVersion nullVersion
     <*> optionalField       "package-name"           basicPkgName
     <*> optionalField       "lib-name"               basicLibName
+  where
+    mkBasic n v pn ln = Basic n v pn (maybe LMainLibName LSubLibName ln)
```

Comment:
`basicLibName` should be of type `LibraryName`

Reply:
it's not exported. wont'fix. (Also `LibraryName` doesn't have `Parsec` instance on purpose, so I'd need to introduce `newtype` etc.)

Reply:
Maybe even using `InstalledPackageIndex` would make sense, even it's not "installed", but "package index" anyway.